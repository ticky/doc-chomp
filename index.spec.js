/* global describe, it, expect */
const DocChomp = require('./index');

describe('Doc Chomp', () => {
  describe('does some basic integrity checkups', () => {
    it('throws when supplied text on the first line', () => {
      expect(() => DocChomp`This shouldn't work`).toThrow();
    });
  });

  describe('makes ethical decisions about your whitespace', () => {
    it('likes space indentation', () => {
      expect(
        /* eslint indent: ["error", 2] */
        DocChomp`
          This document is indented with ${'spaces'}!

          * Spaces are flexible and ${'versatile'}
            * Spaces are ubiquitous
            * Spaces are ${'not'} everything

          These are Doc Chomp's mentor's preferred type of indentation.
          `
        /* eslint indent: ["error", 2] */
      ).toMatchSnapshot();
    });

    it('is open to tab indentation', () => {
      expect(
        /* eslint indent: ["error", "tab"] */
				DocChomp`
					This document is indented with ${'tabs'}!

					* Tabs are ${'uniform'}
						* Tabs adjust to viewer's ${'preferences'}
						* Tabs can mislead when attempting to align to syntax

					What fun.
					`
        /* eslint indent: ["error", 2] */
      ).toMatchSnapshot();
    });
  });
});
