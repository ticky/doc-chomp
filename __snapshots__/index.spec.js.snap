exports[`Doc Chomp makes ethical decisions about your whitespace is open to tab indentation 1`] = `
"This document is indented with tabs!

* Tabs are uniform
	* Tabs adjust to viewer\'s preferences
	* Tabs can mislead when attempting to align to syntax

What fun.
"
`;

exports[`Doc Chomp makes ethical decisions about your whitespace likes space indentation 1`] = `
"This document is indented with spaces!

* Spaces are flexible and versatile
  * Spaces are ubiquitous
  * Spaces are not everything

These are Doc Chomp\'s mentor\'s preferred type of indentation.
"
`;
